src/net_processing.cpp

```diff
@@ -1127,7 +1127,7 @@ PeerLogicValidation::PeerLogicValidation(CConnman* connmanIn, BanMan* banman, CS
     // combine them in one function and schedule at the quicker (peer-eviction)
     // timer.
     static_assert(EXTRA_PEER_CHECK_INTERVAL < STALE_CHECK_INTERVAL, "peer eviction timer should be less than stale tip check timer");
-    scheduler.scheduleEvery([&] { this->CheckForStaleTipAndEvictPeers(consensusParams); }, std::chrono::seconds{EXTRA_PEER_CHECK_INTERVAL});
+    scheduler.scheduleEvery([this, consensusParams] { this->CheckForStaleTipAndEvictPeers(consensusParams); }, std::chrono::seconds{EXTRA_PEER_CHECK_INTERVAL});
```

Comment:
I think this can  be made `=` to just mirror what `std::bind` did. (`this` is a pointer, so it shouldn't matter whether it is copied or referenced, and for consensusParams we want it to be copied).

```suggestion
    scheduler.scheduleEvery([=] { this->CheckForStaleTipAndEvictPeers(consensusParams); }, std::chrono::seconds{EXTRA_PEER_CHECK_INTERVAL});
```

Reply:
The `=` is also what is recommended in the "Tutorial" (written by me, so count that against me):

https://github.com/bitcoin/bitcoin/blob/3a8d25064e700ff2e69600cc1ede597751283a85/src/scheduler.h#L28

Reply:
Doh! I was about to apply this suggestion, but then realized that [The implicit capture of *this when the capture default is = is deprecated. (since C++20)](https://en.cppreference.com/w/cpp/language/lambda#Lambda_capture) :-/ Indeed:

```cpp
        auto m() {
            return [=](){
                std::cout << "lambda: this=" << this << std::endl;
            };
        }
```

Produces:
```
clang++10 -std=c++20 -Wdeprecated t.cc -o t
t.cc:96:49: warning: implicit capture of 'this' with a capture default of '=' is deprecated
      [-Wdeprecated-this-capture]
                std::cout << "lambda: this=" << this << std::endl;
                                                ^
t.cc:95:21: note: add an explicit capture of 'this' to capture '*this' by reference
            return [=](){
                    ^
                     , this
```

Better not add code that would have to be fixed later, so I would rather keep it as is.

What do you think?

PS `this` is always captured by reference and I think all of the following do the same wrt `this`: `[=]`, `[this]`, `[&]`, `std::bind`

Reply:
In that case I'd prefer to use `[=, this]`, but this might not compile either on our current C++ target? According to the page you linked:

```cpp
    [=, this] {};   // until C++20: Error: this when = is the default
                    // since C++20: OK, same as [=]
```

So, given that we won't be switching to C++20 any time soon, I'd still prefer the `[=]`. Then it can be changed, if and when we switch to C++20?

Just a nit, though. ACK from me either way.